### PR TITLE
Remove obsolete loop over null safe/unsafe compiler in test and compiler var into group() scope

### PR DESCRIPTION
**Pull request by timmaffett**
This PR removes the obsolete loop over nullSafety true/false - as the variable was never used and the same tests were repeated with no differences.

Changes:
  - The `compiler` variable is moved within group() scope
  - and helper function `generateCompilerDDCTest` is moved within the group() scope (where setupAllI()/tearDownAll() are going to create and dispose() of the `compiler` variable it operares on).
  - the `for (final nullSafety in [false, true]) {` loop is removed
  - the group() name is changed to hard coded '(Always) Null Safe Compiler'

(That it but the diff looks like a lot because it's lost sync and the change in indenting on everything didn't help. )

### CODE REVIEW

**Comment by domesticmouse** [comment]
Looks like the branch is out of date, wanna bring it up to spec?

File: test/compiler_test.dart [hunk 1]
```diff
@@ -14,195 +14,190 @@ import 'package:test/test.dart';
...
+      expect(result.problems.length, 1);
+      expect(result.problems[0].toString(), contains('Error: Expected'));
+    });
+
+    test('good import', () async {
+      const code = '''
 import 'dart:html';
 
 void main() {
-  var count = querySelector('#count');
```
Comment (domesticmouse) [inline test/compiler_test.dart:149]: Revert space deletion please

**Review by domesticmouse — APPROVED** [review]
Minor nit, and bring branch up to date

File: test/compiler_test.dart (hunk 1 above)
Comment (timmaffett) [reply test/compiler_test.dart]: sorry I didn't catch that. I un-indented that huge code block and that inadvertently got caught up in it.

Branch up to date and indenting fixed.

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**